MANUAL.md

```diff
@@ -0,0 +1,99 @@
+
+
+## Features
+
+## SSL Certificates
+
+* `self-signed-certs` - If you wish to have Genesis generate
+  self-signed certs for you. 
+* `provided-cert` - If you have SSL cert/key to provide, which is
+  grabbed from Vault via path:
+  `$GENESIS_VAULT_PATH/nginx/ssl_certificate:certificate` and
+  `$GENESIS_VAULT_PATH/nginx/ssl_certificate:key`
+
+## Authentication
+
+* `http-auth` - By default, only Grafana supports authentication via
+  the web UI. This feature adds HTTP Basic Authentication to
+  AlertManager and Prometheus with automatically generated
+  credentials.
+
+## Monitoring
+
+* `monitor-cf` - Have Prometheus connect to the CF Firehose to track
+  CF app status + more. Requires two UAA accounts, one with the
+  `cloud_controller.admin_read_only` scope, and the `doppler.firehose`
+  scope. To create these accounts, here's an example:
+```
+uaac client add prometheus-cf \
+  --name prometheus-cf \
+  --secret <64 char secret> \
+  --authorized_grant_types client_credentials,refresh_token \
+  --authorities cloud_controller.admin_read_only
+
+uaac client add prometheus-firehose \
+  --name prometheus-firehose \
+  --secret <64 char secret> \
+  --authorized_grant_types client_credentials,refresh_token \
+  --authorities doppler.firehose
+```
+
+  This will add the user `prometheus-cf` and `prometheus-firehose`,
+  with the secrets of your choice. `prometheus-cf` will have the
+  `cloud_controller.admin_read_only` scope, and the
+  `prometheus-firehose` will have the `doppler.firehose` scope.
+
+  These credentials are stored in `$GENESIS_VAULT_PATH/cf_uaa_logins`:
+
+  *cloud_controller.admin_read_only* scope:
+  * `cf_uaa_logins:cf_exporter_client_id`
+  * `cf_uaa_logins:cf_exporter_client_secret`
+
+  *doppler.firehose* scope:
+  * `cf_uaa_logins:firehose_exporter_client_id`
+  * `cf_uaa_logins:firehose_exporter_client_secret`
+
+
+## Params
+
+### General Infrastructure Configuration
+* `disk_type` - The `persistent_disk_type` that Prometheus should use
+  for storage. (default: `prometheus`)
+* `vm_type`- The `vm_type` that Prometheus should be deployed on.
+  (default: `default`) 
+* `network` - The `network` that Prometheus should be deployed on.
+  (default: `prometheus`)
+* `stemcell_os` - The operating system stemcell you want to deploy on.
+  (default: `ubuntu-trusty`)
+* `stemcell_version` - The specific version of the stemcell you want
+  to deploy on. (default: `latest`)
+* `static_ip` - The static IP to assign to the VM. (no default)
+
+### Prometheus Related Configuration
+* `prometheus_port` - The port for Nginx to use to reverse proxy to
+  Prometheus. (default: `443`)
+* `grafana_port` - The port for the Nginx to use to reverse proxy to
+  Grafana. (default: `8080`)
+* `alertmanager_port` - The port for the Nginx to use to reverse proxy
+  to AlertManager. (default: `8082`)
+* `external_domain` - The domain used to access this Prometheus
+  deployment. Can be either a FQDN or IP address. (no default)
```

Comment:
Can/should this default to the static_ip chosen?

Reply:
So this was one of the dilemmas I often face with `self-signed-cert` features. I can't know in advance the IP of the VM to sign the cert (so we prompt the user for a static IP). But, it's possible an operator wants to access the various webUIs with a FQDN; so the cert will be signed by both a static IP and some currently unknown FQDN.

However, the `kit.yml` needs to know what names to sign for in advance so our YAML looks like this:
```
certificates:
  self-signed-cert:
    nginx:
      ca: { valid_for: 1y }
      ssl_certificate:
        valid_for: 1y
        names:
        - ${params.static_ip}
        - ${params.external_domain}
```

So both `params.static_ip` and `params.external_domain` _must_ be defined within the environment file. I did this because I wanted the cert signed for both the IP AND external domain if an external domain is present. 

I can cut this down and only have `params.external_domain` be signed for the cert (defaults to static IP address if not defined), but in that case the certificate will not be signed for the IP address, only the FQDN. 

In the end, everyone is going to `-k` this endpoints, so perhaps having the cert signed for both is useless. What do you think?

Reply:
I think as soon as you have a domain name, you should also bring your own cert.  If that means you use `safe x509 issue` to generate a new self-signed cert in the Vault, fine.